src/main/java/org/elasticsearch/plugin/readonlyrest/acl/blocks/rules/impl/SearchlogSyncRule.java

```diff
@@ -17,48 +17,22 @@
 
 package org.elasticsearch.plugin.readonlyrest.acl.blocks.rules.impl;
 
-import org.apache.logging.log4j.Logger;
-import org.elasticsearch.action.ActionRequest;
-import org.elasticsearch.action.ActionResponse;
-import org.elasticsearch.action.search.SearchRequest;
-import org.elasticsearch.action.search.SearchResponse;
-import org.elasticsearch.common.bytes.BytesReference;
-import org.elasticsearch.common.logging.Loggers;
-import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.common.xcontent.XContentHelper;
-import org.elasticsearch.plugin.readonlyrest.acl.blocks.BlockExitResult;
 import org.elasticsearch.plugin.readonlyrest.acl.blocks.rules.RuleExitResult;
-import org.elasticsearch.plugin.readonlyrest.acl.blocks.rules.RuleNotConfiguredException;
 import org.elasticsearch.plugin.readonlyrest.acl.blocks.rules.SyncRule;
-import org.elasticsearch.plugin.readonlyrest.wiring.requestcontext.RequestContext;
-
-import java.io.IOException;
-import java.util.Arrays;
-import java.util.Optional;
+import org.elasticsearch.plugin.readonlyrest.requestcontext.RequestContext;
+import org.elasticsearch.plugin.readonlyrest.settings.rules.SearchlogRuleSettings;
 
 /**
  * Created by sscarduzio on 27/03/2017.
  */
 public class SearchlogSyncRule extends SyncRule {
-  private final Logger logger = Loggers.getLogger(getClass());
-
-  private boolean shouldLog = false;
 
-  public SearchlogSyncRule(Settings s) throws RuleNotConfiguredException {
-    try {
-      shouldLog = s.getAsBoolean(getKey(), false);
-      if (!shouldLog) throw new RuleNotConfiguredException();
-    } catch (Exception e) {
-      throw new RuleNotConfiguredException();
-    }
-  }
+  private final boolean shouldLog;
+  private final SearchlogRuleSettings settings;
 
-  public static Optional<SearchlogSyncRule> fromSettings(Settings s) {
-    try {
-      return Optional.of(new SearchlogSyncRule(s));
-    } catch (RuleNotConfiguredException ignored) {
-      return Optional.empty();
-    }
+  public SearchlogSyncRule(SearchlogRuleSettings s) {
+    shouldLog = s.isEnabled();
```

Comment:
Could you explain what is the advantage of separating the match logic from the settings? Now for each rule we have two classes.. There must be some advantage I'm not seeing